src/plugins/intel_gna/src/pre_post_process/transposition_info.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "transposition_info.hpp"
+
+#include <vector>
+
+#include "log/debug.hpp"
+#include "openvino/core/model.hpp"
+#include "openvino/core/shape.hpp"
+#include "openvino/opsets/opset9.hpp"
+
+namespace ov {
+namespace intel_gna {
+namespace pre_post_process {
+
+using namespace ov::opset9;
+
+/*
+ * Convert transposition info to preprocessing model using Transpose layer
+ */
+std::shared_ptr<ov::Model> ToProcessModel(const TranspositionInfo& t_info) {
+    int32_t c_size = t_info.num_transpose_rows;
+    int32_t hw_size = t_info.num_transpose_columns;
+
+    if (!t_info.transpose) {
+        return nullptr;
+    }
+
+    ov::PartialShape input_shape{1, c_size, hw_size};
+    auto param = std::make_shared<Parameter>(ov::element::f32, input_shape);
+
+    // legacy way was to revert C and HW dimentions in the reshaped tensor
+    std::vector<int32_t> reshape_pattern{-1, c_size, hw_size};
+    auto reshape_const =
+        std::make_shared<Constant>(ov::element::i32, ov::Shape{reshape_pattern.size()}, reshape_pattern);
+    auto reshape = std::make_shared<Reshape>(param, reshape_const, false);
+
+    // NCHW -> NHWC or NHWC -> NCHW
+    std::vector<int8_t> transpose_order{0, 2, 1};
+    auto transpose_const =
+        std::make_shared<Constant>(ov::element::i8, ov::Shape{transpose_order.size()}, transpose_order);
+    auto transpose = std::make_shared<Transpose>(reshape, transpose_const);
+
+    auto result = std::make_shared<Result>(transpose);
+
+    std::shared_ptr<ov::Model> model =
```

Comment:
Not clear why we need model containing only params, reshape and transpose

Reply:
to execute it before inference and after